util/purl/image_test.go

```diff
@@ -91,7 +91,6 @@ func TestRefToPURL(t *testing.T) {
 func TestPURLToRef(t *testing.T) {
 	testDgst := digest.FromBytes([]byte("test")).String()
 	p := platforms.Normalize(platforms.DefaultSpec())
-	p.OSVersion = "" // OSVersion is not supported in PURL
```

Comment:
I removed this "reset", per https://github.com/moby/buildkit/pull/4301#issuecomment-1757172771

But I'm somewhat wondering if that's "correct"; are package-URLs meant to be a canonical URL for a specific platform, and if so, should (could) they be os-version specific? (and include the os-version?)

Looking at https://github.com/containerd/containerd/pull/8778 again, I'm starting to wonder if that change was the right thing to do (although, admitted, "Parse" was already filling in missing information). The main issue is that it does not provide for a os-version element, but now _does_ "dream up" one, and it somewhat feels like the paints shut the option of _adding_ an element for os-version in the string-representation.

I wonder if `Parse` and `<insert defaults>` should be separate things, but maybe that ship already sailed when it was decided to "only _architecture_ OR _platform_ are required, everything else being optional".

Reply:
Yeah, I don't see at all how containerd/containerd#8778 could be correct. The `Parse()` converts from input string to struct, but with that PR it doesn't use the input string, but the properties of the machine that does the parsing, which is completely unrelated.

Same for this test. If the input for the test is empty `OSVersion` then one should not just magically appear from purl conversion.

Reply:
I was considering doing the reverse, and update the Purl function to explicitly reset the os-version field; but that's assuming the string format would never support an os-version field. It would perhaps be _slightly_ more correct though; WDYT?

Reply:
@tonistiigi @jedevc I updated the PR, and updated `PURLToRef` to explicitly exclude the `OSVersion` (and `OSFeatures`) from the result.